test/parallel/test-http-client-default-headers-exist.js

```diff
@@ -61,15 +57,14 @@ const server = http.createServer(common.mustCall((req, res) => {
     expectedHeaders[req.method].length,
     `some headers were missing for method: ${req.method}`
   );
-
-  countdown.dec();
 }, expectedMethods.length));
 
 server.listen(0, common.mustCall(() => {
-  expectedMethods.forEach((method) => {
-    http.request({
+  Promise.all(expectedMethods.map(async (method) => {
```

Comment:
It's not an issue with this particular test, but keep in mind that changing all of these to use Promises is changing the timing of these tests... rather than executing the countdown cleanup logic synchronously it is being deferred to the microtask queue. That could end up causing the test to miss issues that may exist when closing things down synchronously so worth watching... also, keep [this issue](https://github.com/nodejs/node/issues/32431) in mind also when refactoring these :-) .. as I said, not an issue on this particular test but there are some where it may be relevant.

Reply:
I want to make sure I understand your point about the cleanup code and the microtask queue. Are you saying that with `countdown`, the callback that checks things executes synchronously whenever the pertinent `.dec()` call happens, whereas with Promise.all(), it is whenever the microtask queue runs the resolve callback in the `.then()` call, so it might execute at a different time and with things in a different state than `countdown`?